ui/apps/platform/src/Containers/BaseImages/BaseImagesPage.tsx

```diff
@@ -1,11 +1,135 @@
+import { useCallback, useState } from 'react';
+import { isAxiosError } from 'axios';
+import {
+    Alert,
+    Bullseye,
+    Button,
+    Divider,
+    Flex,
+    FlexItem,
+    PageSection,
+    Spinner,
+    Text,
+    Title,
+} from '@patternfly/react-core';
+
+import {
+    addBaseImage as addBaseImageFn,
+    deleteBaseImage as deleteBaseImageFn,
+    getBaseImages,
+} from 'services/BaseImagesService';
+import type { AddBaseImageRequest } from 'services/BaseImagesService';
+import useRestMutation from 'hooks/useRestMutation';
+import useRestQuery from 'hooks/useRestQuery';
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+
+import BaseImagesModal from './BaseImagesModal';
+import BaseImagesTable from './BaseImagesTable';
 
 function BaseImagesPage() {
+    const [isAddModalOpen, setIsAddModalOpen] = useState(false);
+
+    const baseImagesRequest = useRestQuery(useCallback(getBaseImages, []));
+
+    const addBaseImageMutation = useRestMutation(
+        (request: AddBaseImageRequest) => addBaseImageFn(request),
+        {
+            onSuccess: () => {
+                baseImagesRequest.refetch();
+            },
+        }
+    );
+
+    const deleteBaseImageMutation = useRestMutation((id: string) => deleteBaseImageFn(id), {
+        onSuccess: () => {
+            baseImagesRequest.refetch();
+        },
+    });
+
+    function handleCloseAddModal() {
+        setIsAddModalOpen(false);
+    }
+
+    function handleOpenAddModal() {
+        addBaseImageMutation.reset();
+        setIsAddModalOpen(true);
+    }
+
+    function handleRemoveBaseImage(id: string) {
+        deleteBaseImageMutation.mutate(id);
+    }
+
+    const baseImages = baseImagesRequest.data ?? [];
+
     return (
-        <PageSection variant="light">
-            <Title headingLevel="h1">Base Images</Title>
-            {/* Page content will be implemented later */}
-        </PageSection>
+        <>
+            <PageSection variant="light">
+                <Flex
+                    alignItems={{ default: 'alignItemsCenter' }}
+                    spaceItems={{ default: 'spaceItemsLg' }}
+                >
+                    <FlexItem flex={{ default: 'flex_1' }}>
+                        <Title headingLevel="h1">Base Images</Title>
+                        <Text>
+                            Manage approved base images for vulnerability tracking and
+                            layer-specific filtering
+                        </Text>
+                    </FlexItem>
+                    <FlexItem align={{ default: 'alignRight' }}>
+                        <Button variant="primary" onClick={handleOpenAddModal}>
+                            Add base image
+                        </Button>
+                    </FlexItem>
+                </Flex>
+            </PageSection>
+            <Divider component="div" />
+            <PageSection className="pf-v5-u-display-flex pf-v5-u-flex-direction-column">
```

Comment:
Are these class names redundant with the only child being a `<Flex>` component? If not, is there an alternative we can use from PatternFly's layout components to avoid the utility classes?
```suggestion
            <PageSection>
```

Reply:
Hmm, trying to remember what I was doing last night. Removing this does prevent the Bullseye from being centered in that content area. I can try a different approach that makes more sense. Actually, I did remember the earlier conversation you had with Mark about potentially using Layout Components vs. classes. I'll try that out and share what worked 👍🏼